controllers/provisioners/eks/helpers.go

```diff
@@ -350,34 +380,49 @@ func (ctx *EksInstanceGroupContext) GetLabelList() []string {
 		isOverride = true
 		overrideLabels := strings.Split(val, ",")
 		for _, label := range overrideLabels {
-			labelList = append(labelList, label)
+			keyVal := strings.Split(label, "=")
+			labelMap[keyVal[0]] = keyVal[1]
 		}
 	}
 
 	if !isOverride {
 		// add default labels
-		labelList = append(labelList, fmt.Sprintf(RoleNewLabelFmt, instanceGroup.GetName()))
+		labelMap[RoleNewLabel] = instanceGroup.GetName()
 
 		// add the old style role label if the cluster's k8s version is < 1.16
 		clusterVersion := ctx.DiscoveredState.GetClusterVersion()
 		ver, err := semver.NewVersion(clusterVersion)
 		if err != nil {
 			ctx.Log.Error(err, "Failed parsing the cluster's kubernetes version", "instancegroup", instanceGroup.GetName())
-			labelList = append(labelList, fmt.Sprintf(RoleOldLabelFmt, instanceGroup.GetName()))
+			labelMap[fmt.Sprintf(RoleOldLabel, instanceGroup.GetName())] = ""
 		} else {
 			c, _ := semver.NewConstraint("< 1.16-0")
 			if c.Check(ver) {
-				labelList = append(labelList, fmt.Sprintf(RoleOldLabelFmt, instanceGroup.GetName()))
+				labelMap[fmt.Sprintf(RoleOldLabel, instanceGroup.GetName())] = ""
 			}
 		}
 	}
 
 	if configuration.GetSpotPrice() == "" {
-		labelList = append(labelList, fmt.Sprintf(InstanceMgrLabelFmt, "lifecycle", v1alpha1.LifecycleStateNormal))
+		labelMap[InstanceMgrLifecycleLabel] = v1alpha1.LifecycleStateNormal
 	} else {
-		labelList = append(labelList, fmt.Sprintf(InstanceMgrLabelFmt, "lifecycle", v1alpha1.LifecycleStateSpot))
+		labelMap[InstanceMgrLifecycleLabel] = v1alpha1.LifecycleStateSpot
 	}
 
+	return labelMap
+}
+
+func (ctx *EksInstanceGroupContext) GetLabelList() []string {
+	var (
+		labelList []string
+	)
+
+	for key, value := range ctx.GetComputedLabels() {
+		if value == "" {
+			value = "\"\""
+		}
+		labelList = append(labelList, fmt.Sprintf("%v=%v", key, value))
```

Comment:
What happens if value is `""`? is it rejected by API?

Reply:
This conditional was to provide compatibility with existing behavior under test. I'm not sure w.r.t. it being rejected.